File: works/2705.java
import java.util.*;

class Main
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        int[] d = new int[1010];
        d[1] = 1;
        d[2] = 2;

        for(int i = 1; i <= 1000; i++)
        {
            int sum = 0;
            for(int j = 1; j <= i/2; j++) sum += d[j];
            d[i] = sum + 1;
        }

        int t = sc.nextInt();
        while(t-- > 0)
        {
            int n = sc.nextInt();
            System.out.println(d[n]);
        }
    }

}

File: works/2705.kt
import java.util.*

fun main()
{
    val sc = Scanner(System.`in`)
    var d = Array<Int>(1010){0}
    d[1] = 1
    d[2] = 2

    for(i in 1..1000)
    {
        var sum = 0
        for(j in 1..i/2) sum += d[j]
        d[i] = sum + 1
    }

    var t = sc.nextInt()
    while(t-- > 0)
    {
        var n = sc.nextInt()
        println(d[n])
    }
}